### PR TITLE
Core: Fix return RTC flag, Events Disable flag

### PR DESCRIPTION
**Pull request by jmcardon**
Rename progn to do
```
pact>(do 1 2 3)
3
```
Factor out common code between CEK/Direct

PR checklist:

* [x] Test coverage for the proposed changes
* [x] PR description contains example output from repl interaction or a snippet from unit test output

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -2,10 +2,11 @@ PR checklist:
 
 * [ ] Test coverage for the proposed changes
 * [ ] PR description contains example output from repl interaction or a snippet from unit test output
-* [ ] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/kadena-io/pact-core/blob/master/CHANGELOG.md)
+* [ ] (If Relevant) New builtins have a FV translation
```
Comment (rsoeldner) [inline .github/pull_request_template.md:5]:
```suggestion
```

File: pact/Pact/Core/IR/Eval/Runtime/Utils.hs [hunk 2]
```diff
@@ -71,6 +78,42 @@ import Pact.Core.Guards
...
+  let pe = PactEvent name params pactModule mhash
+  emitEventUnsafe pe
+
+emitEvent
+  :: i
+  -> PactEvent PactValue
+  -> EvalM e b i ()
+emitEvent info pe = findCallingModule >>= \case
+    Just mn -> do
+      -- Todo: ++ definitely feels suboptimal, especially for gas.
```
Comment (rsoeldner) [inline pact/Pact/Core/IR/Eval/Runtime/Utils.hs:93]: consider moving the comment to `emitEventUnsafe`, as `emitEvent` does not use `++`.

File: pact/Pact/Core/IR/Eval/Runtime/Utils.hs [hunk 3]
```diff
@@ -150,6 +193,15 @@ safeTail [] = []
...
 
+whenExecutionFlagSet :: ExecutionFlag -> EvalM e b i () -> EvalM e b i ()
+whenExecutionFlagSet flag act =
+  isExecutionFlagSet flag >>= (`when` act)
+
+
+unlessExecutionFlagSet :: ExecutionFlag -> EvalM e b i () -> EvalM e b i ()
+unlessExecutionFlagSet flag act =
+  isExecutionFlagSet flag >>= (`unless` act)
+
```
Comment (rsoeldner) [inline pact/Pact/Core/IR/Eval/Runtime/Utils.hs:204]: Should these also be `inlineable`?

File: pact/Pact/Core/IR/Eval/Runtime/Utils.hs [hunk 4]
```diff
@@ -186,11 +240,13 @@ asBool info b pv =
   throwExecutionError info (NativeArgumentsError (builtinName b) [pvToArgTypeError pv])
 {-# INLINABLE asBool #-}
 
+-- | Todo: revisit
```
Comment (rsoeldner) [inline pact/Pact/Core/IR/Eval/Runtime/Utils.hs:243]: Leftover

File: pact/Pact/Core/IR/Eval/Runtime/Utils.hs [hunk 5]
```diff
@@ -186,11 +240,13 @@ asBool info b pv =
...
 checkSchema :: M.Map Field PactValue -> Schema -> Bool
 checkSchema o (Schema _ sc) =
   M.size o == M.size sc &&
   getAll (M.foldMapWithKey (\k v -> All $ maybe False (`checkPvType` v) (M.lookup k sc)) o)
 
+-- | Todo: revisit
```
Comment (rsoeldner) [inline pact/Pact/Core/IR/Eval/Runtime/Utils.hs:249]: Leftover

**Review by rsoeldner — APPROVED** [review]

File: pact/Pact/Core/IR/Eval/Runtime/Utils.hs (hunk 3 above)
Comment (jmcardon) [reply pact/Pact/Core/IR/Eval/Runtime/Utils.hs:204]: True 👍

**Pull request merged**